src/BitChopp.Main/ViewModels/MainViewModel.cs

```diff
@@ -59,24 +73,55 @@ private async void OnSwitchSelected(SwitchCommandObject swObj)
         {
             await ShowSuccessWindow(swObj);
         }
+
+        qrWindow.Close();
+        qrWindow = null;
     }
 
     private async Task ShowSuccessWindow(SwitchCommandObject swObj)
     {
         var volume = ExtractVolume(swObj.Switch.Description);
+        _successViewModel = new SuccessViewModel(volume, _configService);
+
+        _pourService.FlowCounterUpdated += (s, e) =>
+        {
+            Dispatcher.UIThread.InvokeAsync(() =>
```

Comment:
Maybe redundant: event is already being dispatched in the UI Thread

Reply:
I had issues with the UI not being updated, so I made sure everything touching UI would go in the UI thread...